pkg/services/ngalert/provisioning/contactpoints.go

```diff
@@ -42,9 +45,15 @@ type ContactPointQuery struct {
 	// Optionally filter by name.
 	Name  string
 	OrgID int64
+	// Optionally decrypt secure settings, requires OrgAdmin.
+	Decrypt bool
 }
 
-func (ecp *ContactPointService) GetContactPoints(ctx context.Context, q ContactPointQuery) ([]apimodels.EmbeddedContactPoint, error) {
+// GetContactPoints returns contact points. If q.Decrypt is true and the user is an OrgAdmin, decrypted secure settings are included instead of redacted ones.
+func (ecp *ContactPointService) GetContactPoints(ctx context.Context, q ContactPointQuery, u *user.SignedInUser) ([]apimodels.EmbeddedContactPoint, error) {
+	if q.Decrypt && (u == nil || !u.HasRole(org.RoleAdmin)) {
```

Comment:
This will not work in GE. Afaik, the legacy roles get deprecated. So, I think it should be a new action (perhaps with no scope), and that action should be assignable to the Grafana admin role. @IevaVasiljeva please correct me if I am wrong.

Reply:
Tested for users with the three basic GE roles: viewer (with added prov. read), writer (with added prov. read), orgadmin. With . Both viewer and writer get 403 when trying to use `decrypt=true`.

Reply:
Thanks for flagging this Yuri! We still provide the basic roles (viewer, editor and admin), so this will currently work. However, the whole point of RBAC is making permission system more flexible, so we should have granular permissions (an action and an optional scope) for all Grafana resources. We also aim to eventually get rid of `HasRole` checks in favour of granular permission checks. So I agree with Yuri that you should avoid checking the basic role here if possible.

Reply:
It would be great if we include it here. I am not sure that the permission check should happen in the service and not API handler. WDYT?

Reply:
I tried to implement it, and it turned out to touch quite a lot of files. So, let's merge it as it is and update in a follow-up PR.